userspace/libsinsp/sinsp_syslog.cpp

```diff
@@ -118,11 +119,11 @@ void sinsp_syslog_decoder::decode_message(const char *data, uint32_t len, char*
 	m_msg.assign(data + pristrlen + 2, len - pristrlen - 2);
 }
 
-const std::string sinsp_syslog_decoder::get_info_line()
+const std::string& sinsp_syslog_decoder::get_info_line()
```

Comment:
I'm not against returning an `std::string` per se. I wasn't actually trying to correct the const value, though I agree that we should remove the const if returning by value. The reason I changed the function to return a reference to const is a warning that emerged in its usage. And there's the rub...

The function is used in `sinsp_filter_check_event::extract()` where `c_str()` is used against the string return by `get_info_line` and then returned as a `uint8_t*`. The net result is returning a pointer to memory owned by a local variable, which Clang correctly detects as dangerous. Using a reference in `get_info_line` was a quick way to avoid this; other solutions are possible, but it's a larger refactoring I did not want to implement, given the scope of the PR.

Please review the reason I just described. If I (or, better, Clang) understood the situation correctly, I think we could go back to the string returned by value, but it needs to go together with a change in the suspicious usages of the function. Given the different nature of this change I'd keep it for another PR.

BTW: I like your change so that a member function whose name starts with `get_` doesn't change the object and I'd suggest to make the function const as well.